test/tools/router_check/router.cc

```diff
@@ -76,43 +71,31 @@ bool RouterCheckTool::compareEntriesInJson(const std::string& expected_route_jso
     }
     Json::ObjectSharedPtr validate = check_config->getObject("validate");
 
-    const std::unordered_map<std::string, std::function<bool(ToolConfig&, const std::string&)>>
-        checkers = {
-            {"cluster_name",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareCluster(tool_config, expected);
-             }},
-            {"virtual_cluster_name",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareVirtualCluster(tool_config, expected);
-             }},
-            {"virtual_host_name",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareVirtualHost(tool_config, expected);
-             }},
-            {"path_rewrite",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareRewritePath(tool_config, expected);
-             }},
-            {"host_rewrite",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareRewriteHost(tool_config, expected);
-             }},
-            {"path_redirect",
-             [this](ToolConfig& tool_config, const std::string& expected) -> bool {
-               return compareRedirectPath(tool_config, expected);
-             }},
-        };
+    using checkerFunc = std::function<bool(ToolConfig&, const std::string&)>;
+    const std::unordered_map<std::string, checkerFunc> checkers = {
+        {"cluster_name",
+         [this](auto&... params) -> bool { return this->compareCluster(params...); }},
+        {"virtual_cluster_name",
+         [this](auto&... params) -> bool { return this->compareVirtualCluster(params...); }},
+        {"virtual_host_name",
+         [this](auto&... params) -> bool { return this->compareVirtualHost(params...); }},
+        {"path_rewrite",
+         [this](auto&... params) -> bool { return this->compareRewritePath(params...); }},
+        {"host_rewrite",
+         [this](auto&... params) -> bool { return this->compareRewriteHost(params...); }},
+        {"path_redirect",
+         [this](auto&... params) -> bool { return this->compareRedirectPath(params...); }},
+    };
 
     // Call appropriate function for each match case
-    for (std::pair<std::string, std::function<bool(ToolConfig&, std::string)>> test : checkers) {
+    for (const auto& test : checkers) {
       if (validate->hasObject(test.first)) {
-        std::string expected = validate->getString(test.first);
+        const std::string& expected = validate->getString(test.first);
```

Comment:
Are you storing a reference to a temporary here?

Reply:
Yeah, because I through it made it clearer what exactly was being passed into the checker functions.